src/drivers/gps/gps.cpp

```diff
@@ -381,7 +404,10 @@ GPS::task_main()
 					_report_gps_pos.epv = 10000.0f;
 					_report_gps_pos.fix_type = 0;
 
-					if (!(_pub_blocked)) {
+					/* update mavlink hil gps parameter */
+					update_param_hil_gps();
```

Comment:
@stmoon  
> I thought parameter should be updated every time because it can be changed anytime.

Yes, but you don't need to call param_find() for that every time, which is expensive. So just run find() in the init, and then call updated()/get() to get the updated param.